qa/workunits/rbd/krbd_watch_errors_exclusive.sh

```diff
@@ -0,0 +1,31 @@
+#!/usr/bin/env bash
+
+set -ex
+set -o pipefail
+
+readonly IMAGE_NAME="watch-errors-exclusive-test"
+
+rbd create -s 1G --image-feature exclusive-lock,object-map "${IMAGE_NAME}"
+
+# induce a watch error every 30 seconds
+dev="$(sudo rbd device map -o exclusive,osdkeepalive=60 "${IMAGE_NAME}")"
+dev_id="${dev#/dev/rbd}"
+
+sudo dmesg -C
+
+# test that a workload doesn't encounter EIO errors
+fio --name test --filename="${dev}" --ioengine=libaio --direct=1 \
+    --rw=randwrite --norandommap --randrepeat=0 --bs=512 --iodepth=128 \
+    --time_based --runtime=1h --eta=never
+
+num_errors="$(dmesg | grep -c "rbd${dev_id}: encountered watch error")"
+echo "Recorded ${num_errors} watch errors"
+
+sudo rbd device unmap "${dev}"
+
+if ((num_errors < 60)); then
```

Comment:
Why was number 60 picked? So you expect the watch error to be detected at least 60 times by the kclient in 1 hour when the watch error is induced every 30 seconds?

Reply:
Any number bigger than 30 would have done.  When dealing with time/timeouts, I like to go in factors, which usually means doubling.

Reply:
(I edit my previous answer substantially going into more detail, please take another look if you have already.)

Reply:
Thanks for the edited answer. :) This is easier to follow. I previously didn't get the bit about even testing for at least 30 is sufficient.